pkg/manifests/manifests_test.go

```diff
@@ -1419,6 +1419,30 @@ func TestRemoteWriteAuthorizationConfig(t *testing.T) {
 	}
 }
 
+func TestPrometheusK8sHttpConfig(t *testing.T) {
```

Comment:
Tested here and with curl. Not sure if it is needed to add some e2e test.
`Allow: GET, OPTIONS, POST
Content-Length: 19
Content-Security-Policy: frame-ancestors 'none'
Content-Type: text/plain; charset=utf-8
Date: Fri, 23 Jun 2023 13:08:26 GMT
Gap-Auth: system:serviceaccount:openshift-monitoring:prometheus-k8s@cluster.local
Gap-Upstream-Address: localhost:9090
X-Content-Type-Options: nosniff`

Reply:
Given that there are several services that require the changes and that the configuration isn't uniform, an e2e test would be useful.

Reply:
If we implement e2e tests, I wouldn't require unit tests as they feel redundant.

Reply:
done